rfcs/text/0000-multi-data-collection-dataset-processing.md

```diff
@@ -0,0 +1,115 @@
+### DCP PR:
+
+***Leave this blank until the RFC is approved** then the **Author(s)** must create a link between the assigned RFC number and this pull request in the format:*
+
+`[dcp-community/rfc#](https://github.com/HumanCellAtlas/dcp-community/pull/<PR#>)`
+
+# RFC: Processing Datasets that Span Multiple Bundles
+
+## Summary
+This RFC proposes a solution to allow processing of datasets that span multiple bundles. It addresses the general problem that the current DCP data model contains no representation of data set grouping and version-completeness beyond that of individual data bundles.  This results in a restriction that all data processing within the DCP is a one-to-one linear set of steps deriving one bundle from another.
+
+However, there are use cases where a giving processing step can require input from multiple input bundles,
+resulting in a full DAG rather than a linear graph. The DAG sub-graph must be defined and all data available to initiate processing.  A notion of data completeness is required to support the implementation of DAG processing.
+
+This RFC defines a general mechanism for grouping bundles, called *data groups*, to allow bundles to be grouped in new ways.   By providing a general method for grouping bundles, this proposal provides a mechanism for addressing various tasks that cross bundle boundaries.  This impacts analysis pipelines, algorithmic complexity, data consistency, data set quality control and data releases.  The current bundle grouping under consideration is a submission to a project.  However, the concept generalizes in a manner that other groupings can be defined to accommodate future needs.
```

Comment:
Interesting question.  I think we need to go through some of the downstream use cases and see what is easier for the consumers.  It is possible.

If it is just the files/bundles in an update, then analysis would query outside of the bounds of the data group to determine what needs to processed.   If the DG corresponds to something larger that bounds what needs to get processed it might make the query easier.

My feeling is that the DG will be more understandable if it corresponds to some scientific entity (experiment, project, project of projects)  and then updates are updates to the DG rather than new DGs